// @sap-ux/adp-tooling/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getVariant = exports.getPromptsForChangeDataSource = exports.getPromptsForChangeInbound = exports.isCFEnvironment = exports.getManifest = exports.promptGeneratorInput = exports.generateChange = exports.migrate = exports.generate = void 0;
__exportStar(require("./types"), exports);
__exportStar(require("./prompts"), exports);
__exportStar(require("./preview/adp-preview"), exports);
__exportStar(require("./base/helper"), exports);
__exportStar(require("./base/validators"), exports);
__exportStar(require("./base/abap"), exports);
var writer_1 = require("./writer");
Object.defineProperty(exports, "generate", { enumerable: true, get: function () { return writer_1.generate; } });
Object.defineProperty(exports, "migrate", { enumerable: true, get: function () { return writer_1.migrate; } });
var editors_1 = require("./writer/editors");
Object.defineProperty(exports, "generateChange", { enumerable: true, get: function () { return editors_1.generateChange; } });
var prompt_1 = require("./base/prompt");
Object.defineProperty(exports, "promptGeneratorInput", { enumerable: true, get: function () { return prompt_1.promptGeneratorInput; } });
var abap_1 = require("./base/abap");
Object.defineProperty(exports, "getManifest", { enumerable: true, get: function () { return abap_1.getManifest; } });
var cf_1 = require("./base/cf");
Object.defineProperty(exports, "isCFEnvironment", { enumerable: true, get: function () { return cf_1.isCFEnvironment; } });
var prompts_1 = require("./prompts");
Object.defineProperty(exports, "getPromptsForChangeInbound", { enumerable: true, get: function () { return prompts_1.getPromptsForChangeInbound; } });
Object.defineProperty(exports, "getPromptsForChangeDataSource", { enumerable: true, get: function () { return prompts_1.getPromptsForChangeDataSource; } });
var helper_1 = require("./base/helper");
Object.defineProperty(exports, "getVariant", { enumerable: true, get: function () { return helper_1.getVariant; } });
